### PR TITLE
Adding loader for TAU NIGENS SSE 2020 dataset

### PR DESCRIPTION
**Pull request by iranroman**
### Description

Please include the following information at the top level docstring for the dataset's module mydataset.py:

- [x] Describe annotations included in the dataset
- [x] Indicate the total duration of the dataset in hours, and (optionally) also list the number of individual files
- [x] Mention the origin of the dataset (e.g. creator, institution)
- [x] Describe the type of audio included in the dataset
- [x] Indicate any relevant papers related to the dataset
- [x] Include a description about how the data can be accessed and the license it uses (if applicable)

#### Dataset loaders checklist:

- [x] Create a script in `scripts/`, e.g. `make_my_dataset_index.py`, which generates an index file.
- [x] Run the script on the canonical version of the dataset and save the index in `soundata/indexes/` e.g. `my_dataset_index.json`.
- [x] Create a module in soundata, e.g. `soundata/my_dataset.py`
- [x] Create tests for your loader in `tests/`, e.g. `test_my_dataset.py`
- [x] Add your module to `docs/source/soundata.rst` and `docs/source/quick_reference.rst`
- [x] Run `tests/test_full_dataset.py` on your dataset.

This PR also addresses issue #75

### CODE REVIEW

**Comment by magdalenafuentes** [comment]
@iranroman can you fix the following mistake in the docs? In `tutorial.rst`:

```
dataset = tf.data.Dataset.from_generator(
    urbansound8k_generator('urbansound8k'),
    {
        "audio": tf.float32,
        "sample_rate": tf.float32,
        "label": tf.string,
        "metadata": {'clip_id': tf.string, 'fold': tf.string}
    }
)
```

change `urbansound8k_generator` for `data_generator`. Thanks!

File: soundata/annotations.py [hunk 1]
```diff
@@ -159,12 +159,16 @@ def validate_lengths_equal(array_list):
     if len(array_list) == 1:
         return
 
-    for att1, att2 in zip(array_list[:1], array_list[1:]):
-        if att1 is None or att2 is None:
-            continue
+    else:
```
Comment (magdalenafuentes) [inline soundata/annotations.py]: I think that you don't need this `else`

File: soundata/annotations.py [hunk 2]
```diff
@@ -159,12 +159,16 @@ def validate_lengths_equal(array_list):
...
+            if att1 is None or att2 is None:
+                continue
 
-        if not len(att1) == len(att2):
-            raise ValueError("Arrays have unequal length")
+            if not len(att1) == len(att2):
+                raise ValueError("Arrays have unequal length")
+
+        # recurse
+        validate_lengths_equal(array_list[:-1])
```
Comment (magdalenafuentes) [inline soundata/annotations.py]: `array_list[1:]`

Comment (magdalenafuentes) [reply soundata/annotations.py]: Also as we discussed offline, we should add a test with a list with three arrays, and e.g. the last one has the wrong length so it triggers a `ValueError`

Comment (magdalenafuentes) [reply soundata/annotations.py]: Thanks for this!

File: soundata/datasets/tau_nigens_sse_2020.py [hunk 3]
```diff
@@ -0,0 +1,641 @@
...
+        room impulse responses (RIRs) of diverse acoustic environments. The sound events are 
+        spatialized as either stationary sound sources, or moving sound sources, in which case 
+        time-variant RIRs are used. 
+        Each scene recording is delivered in microphone array (MIC) and first-order Ambisonics (FOA) 
+        format. 
+    *Audio Files Included:*
+   	* 600 one-minute long sound scene recordings (development dataset).
+   	* 200 one-minute long sound scene recordings (evaluation dataset).
+        * Sampling rate is 24 kHz (16-bit signed integer PCM).
+   	* About 700 sound event samples spread over 14 classes (see here for more details).
```
Comment (magdalenafuentes) [inline soundata/datasets/tau_nigens_sse_2020.py:26]: I guess that the `here` is a link?

File: soundata/datasets/tau_nigens_sse_2020.py [hunk 4]
```diff
@@ -0,0 +1,641 @@
...
+   	* From about 1500 to 3500 possible RIR positions across the different rooms.
+   	* Both static reverberant and moving reverberant sound events.
+   	* Up to two overlapping sound events allowed, temporally and spatially.
+   	* Realistic spatial ambient noise collected from each room is added to the spatialized sound events, at varying signal-to-noise ratios (SNR) ranging from noiseless (30dB) to noisy (6dB).
+    *Annotations Included:*
+        * Each recording in the development set has labels of events and Directions of arrival in a plain csv file with the same filename.
+        * Each row in the csv file has a frame number, active class index, track number index, azimuth, and elevation.
+        * Frame, class, and track enumeration begins at 0. 
+        * Frames correspond to a temporal resolution of 100msec. 
+        * Azimuth and elevation angles are given in degrees, rounded to the closest integer value, with azimuth and elevation being zero at the front, azimuth $\phi \in [-180^{\circ}, 180^{\circ}]$, and elevation $\theta \in [-90^{\circ}, 90^{\circ}]$. Note that the azimuth angle is increasing counter-clockwise ($\phi = 90^{\circ}$ at the left).
```
Comment (magdalenafuentes) [inline soundata/datasets/tau_nigens_sse_2020.py]: The ecuations don't render properly in the docs, maybe you can change them to be markdown friendly or remove them altogether?

File: soundata/datasets/tau_nigens_sse_2020.py [hunk 5]
```diff
@@ -0,0 +1,639 @@
...
+        azimuths_unit (str): azimuths unit, one of AZIMUTHS_UNITS
+        distances (list): list of size n with np.ndarrays with dtype int,
+            indicating the distance of the sound event per time_step if moving
+            or a single value if static. Values must be positive or None
+        distances_unit (str): distances unit, one of DISTANCES_UNITS
+        labels (list): list of event labels (as strings)
+        labels_unit (str): labels unit, one of LABELS_UNITS
+        track_number_indices (list): list of track number indices (as strings)
+        confidence (np.ndarray or None): array of confidence values, float in [0, 1]
+    """
```
Comment (magdalenafuentes) [inline soundata/datasets/tau_nigens_sse_2020.py]: I understand you defined this here because of the conversation in #61, in which we didn't decide on a specific class to add in `annotations` and instead decided to create the classes in each loader instead. But it does fill like a lot of repetition in the code... What about importing the class from the `tau_sse_2019` loader and only adding the two extra attributes here? I think it will make our lives easier in the future when finally moving this classes to `annotations`

File: soundata/datasets/tau_nigens_sse_2020.py [hunk 6]
```diff
@@ -0,0 +1,639 @@
...
+                    np.concatenate(
+                        [e[:, np.newaxis], a[:, np.newaxis], d[:, np.newaxis]], axis=1
+                    )
+                )
+                for e, a, d in zip(els, azs, dis)
+            ]
+            for els, azs, dis in zip(elevations, azimuths, distances)
+        ]
+        [
+            [
```
Comment (magdalenafuentes) [inline soundata/datasets/tau_nigens_sse_2020.py:279]: wow, this formatting is really hard to parse, I'm guessing is `black` who's doing this? let's take it offline and see if we can fix this

File: soundata/datasets/tau_nigens_sse_2020.py [hunk 7]
```diff
@@ -0,0 +1,639 @@
...
+    Raises:
+        ValueError: if intervals have an invalid shape or
+                have cartesian coordinate values outside the expected ranges.
+    """
+
+    # validate that locations have the correct shape
+    locations_shape = np.shape(locations)
+    if len(locations_shape) != 2 or locations_shape[1] != 3:
+        raise ValueError(
+            f"Locations should be arrays with three columns, but array has {locations_shape}"
```
Comment (magdalenafuentes) [inline soundata/datasets/tau_nigens_sse_2020.py]: has `shape`

File: soundata/datasets/tau_nigens_sse_2020.py [hunk 8]
```diff
@@ -0,0 +1,639 @@
...
+    elif len(locations) == 1 and interval[1] - interval[0] > 0:
+        pass  # the event is static
+    # if the object is static, validation passes
+    elif not np.isclose(len(locations) - 1, (interval[1] - interval[0]) / time_step):
+        raise ValueError(
+            "The number of locations does not fit in the interval, given the time_step"
+        )
+
+
+def validate_locations(locations):
```
Comment (magdalenafuentes) [inline soundata/datasets/tau_nigens_sse_2020.py:329]: same comment as the `SpatialEvents` class

File: soundata/datasets/tau_nigens_sse_2020.py [hunk 9]
```diff
@@ -0,0 +1,639 @@
...
+    def audio(self) -> Optional[Tuple[np.ndarray, float]]:
+        """The clip's audio
+        Returns:
+            * np.ndarray - audio signal
+            * float - sample rate
+        """
+        return load_audio(self.audio_path)
+
+    @core.cached_property
+    def events(self) -> Optional[SpatialEvents]:
```
Comment (magdalenafuentes) [inline soundata/datasets/tau_nigens_sse_2020.py]: better call them `spatial_events`?

File: soundata/datasets/tau_nigens_sse_2020.py (hunk 5 above)
Comment (magdalenafuentes) [reply soundata/datasets/tau_nigens_sse_2020.py]: Sorry, now that I look again it seems you need lists of arrays here, so maybe let's sync offline

**Comment by iranroman** [comment]
This one is ready to be reviewed, approved, and merged

**Review by magdalenafuentes — APPROVED** [review]
